templates/todo/api/java/src/main/resources/application.properties

```diff
@@ -3,11 +3,13 @@ server.port=3100
 spring.jackson.date-format=com.microsoft.azure.simpletodo.configuration.RFC3339DateFormat
 spring.jackson.serialization.WRITE_DATES_AS_TIMESTAMPS=false
 
+# Temporally disabling auto-configuration for key vault
 # Import KeyVault secrets as properties
-spring.cloud.azure.keyvault.secret.property-sources[0].enabled=true
-spring.cloud.azure.keyvault.secret.property-sources[0].endpoint=${AZURE_KEY_VAULT_ENDPOINT}
+# spring.cloud.azure.keyvault.secret.property-sources[0].enabled=true
+# spring.cloud.azure.keyvault.secret.property-sources[0].endpoint=${AZURE_KEY_VAULT_ENDPOINT}
+custom.azure.keyvault.secret.endpoint=${AZURE_KEY_VAULT_ENDPOINT},AZURE-COSMOS-CONNECTION-STRING
 
-spring.data.mongodb.uri=${AZURE_COSMOS_CONNECTION_STRING:#{null}}
+#spring.data.mongodb.uri=${AZURE_COSMOS_CONNECTION_STRING:#{null}}
```

Comment:
My understanding (not a java guy) is that instead of this `spring.data.mongodb.uri` to set the mongo endpoint configuration value, the `@Bean` stuff injects logic that configures the Mongo object as it's being constructed.

Does that mean we can just delete this instead of commenting it out?

Reply:
The `@Bean stuff` sets a customizer callback that runs after the auto-configuration (The auto-configuration comes from this application-properties file).  This is the order for creating the mongo client:
- Run the @Bean customizer to set a callback
- Create the mongo configuration settings from the auto-config
- Call the customizer callback that allows overriding the `configuration` from the auto-settings
- Create the mongo client with the settings 

If we don't comment this line, the second step fails, because the `${AZURE_COSMOS_CONNECTION_STRING}` is resolved as an empty string and mongo validate that the connection string is valid